### ISSUE TITLE
wirte_parquet() and write_delta() panics when writing large dataset sizes

### ISSUE THREAD

**Issue by edgBR** [pola-rs/polars#16638]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

Given the following dataset shape:

```python
INFO:__main__:Getting Storage Credentials
INFO:azureml._restclient.clientbase:Created a worker pool for first use
INFO:__main__:Lazy read accounts data
INFO:__main__:Lazy read payments data
INFO:__main__:Getting vendor types table
INFO:__main__:Streaming join and account level features
INFO:__main__:Joined data row count (2573952862, 29)
INFO:__main__:Aggregating data at portfolio level
INFO:__main__:Aggregating data at portfolio and payment channel level
INFO:__main__:Joining channel data, vendor type data and portfolio data
INFO:__main__:Joined data row count (36657598, 41)
INFO:__main__:Writing parquet table
```
When writting:

```python
final_data.write_parquet(
            file="./outputs/portfolio_enhance_data.csv",  # in reality we should write to silver
        )

```

### Log output

```shell
hread '<unnamed>' panicked at crates/polars-core/src/chunked_array/ops/chunkops.rs:95:9:
polars' maximum length reached. Consider installing 'polars-u64-idx'.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Cleaning up all outstanding Run operations, waiting 300.0 seconds
1 items cleaning up...
Cleanup took 0.10357856750488281 seconds
Traceback (most recent call last):
  File "<string>", line 197, in <module>
  File "<string>", line 193, in main
  File "/opt/miniconda/lib/python3.10/runpy.py", line 289, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/opt/miniconda/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/opt/miniconda/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "preprocess/main.py", line 573, in <module>
    run(args)
  File "preprocess/main.py", line 518, in run
    final_data.write_parquet(
  File "/opt/miniconda/lib/python3.10/site-packages/polars/dataframe/frame.py", line 3292, in write_parquet
    self._df.write_parquet(
pyo3_runtime.PanicException: polars' maximum length reached. Consider installing 'polars-u64-idx'.
```

### Issue description

The write parquet operation fails despite being the row size under the limit of 2^32 billion rows in all the datasets using in my job. The dataset Im trying to write is 36M rows The machine Im using to run my data processing pipeline is enough to handle the amount of rows. This one is:

Standard_FX48mds (48 cores, 1008 GB RAM, 2016 GB disk)

### Expected behavior

Code works as expected.

### Installed versions

<details>

Fully reproducible environment. Dockerfile:
```docker

# Start with a base image, for example:
# FROM mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu20.04

# Use the provided environment variables for conda and environment file paths
ARG CONDA_FILE
ARG IMAGE_NAME
FROM mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu20.04

RUN echo "CONDA_FILE: preprocessing.yml"
RUN echo "IMAGE_NAME: mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu20.04"

RUN rm /bin/sh && ln -s /bin/bash /bin/sh
RUN echo "source /opt/miniconda/etc/profile.d/conda.sh && conda activate" >> ~/.bashrc

COPY preprocessing.yml conda_env.yml

RUN cat conda_env.yml

RUN source /opt/miniconda/etc/profile.d/conda.sh && \
    conda activate && \
    conda install conda && \
    pip install cmake && \
    conda env update -f conda_env.yml

```

Conda yaml:

```yaml
channels:
  - defaults
dependencies:
  - pip=24.0=py310h06a4308_0
  - python=3.10.14=h955ad1f_1
  - zlib=1.2.13=h5eee18b_1
  - pip:
    - azureml-sdk==1.56.0
    - matplotlib==3.9.0
    - pandas==2.2.2
    - polars==0.20.30
    - pyarrow==16.1.0
    - fsspec==2024.5.0
    - deltalake==0.17.4
  - conda:
    - conda=23.5.0
```

</details>

**Comment by ritchie46** [comment]
Are you entirely sure it is that amount of rows? Have you tried `polars-u64-idx`?

**Comment by edgBR** [comment]
Hi @ritchie46 

Yes, I am entirely sure.

The data I'm writing is the one corresponding to :

`INFO:__main__:Joined data row count (36657598, 41)`

With regards to installing polars-u64-idx. The error persists even I install the package.

**Comment by edgBR** [comment]
Hi,

Small update, I think I have narrowed down the issue.

It seems one of my aggregations did not work as expected due to a bug in the data. This resulted in the creation of a list type column with considerable size. Concretely every row of the aggregation had this list column of average size 127750 elements. As my data frame has 36.5M rows this might have been the case to this warning.

The code works perfectly fine when I removed this columns.

Would it be possible to keep this open? Even that the issue is probably not related with the delta writer I found the exception message a bit confusing.
